src/scripts/main.js

```diff
@@ -1,3 +1,64 @@
 'use strict';
 
-// write code here
+const appendRowButton = document.querySelector('.append-row');
+const removeRowButton = document.querySelector('.remove-row');
+const appendColumnButton = document.querySelector('.append-column');
+const removeColumnButton = document.querySelector('.remove-column');
+const table = document.querySelector('.field');
+
+const MAX_SIZE = 10;
+const MIN_SIZE = 2;
+
+function updateButtons() {
+  const rowCount = table.rows.length;
+  const colCount = table.rows[0].cells.length;
+
+  appendRowButton.disabled = rowCount >= MAX_SIZE;
+  removeRowButton.disabled = rowCount <= MIN_SIZE;
+
+  appendColumnButton.disabled = colCount >= MAX_SIZE;
+  removeColumnButton.disabled = colCount <= MIN_SIZE;
+}
+
+appendRowButton.addEventListener('click', () => {
+  if (table.rows.length < MAX_SIZE) {
+    const newRow = table.insertRow();
+    const cellCount = table.rows[0].cells.length;
+
+    for (let i = 0; i < cellCount; i++) {
+      newRow.insertCell();
+    }
+    updateButtons();
+  }
+});
+
+removeRowButton.addEventListener('click', () => {
+  if (table.rows.length > MIN_SIZE) {
+    table.deleteRow(-1);
+    updateButtons();
+  }
+});
+
+appendColumnButton.addEventListener('click', () => {
+  const colCount = table.rows[0].cells.length;
```

Comment:
Ensure that the table has at least one row and one cell initially to avoid errors when accessing `table.rows[0].cells.length`. If the table is empty, this line will throw an error.